src/main/java/database/DataBaseService.java

```diff
@@ -28,7 +32,7 @@ public void saveTemperature(Temperature temperature) {
 			ps.setTimestamp(1, tDate);
 			ps.setDouble(2, temperature.getValue());
 			ps.setString(3, temperature.getType().toString());
-			ps.execute();			
+			ps.execute();
 			conn.close();
```

Comment:
`conn.close();` liec iekš final bloka.
Šajā gadījumā var būt problēma, ja pēkšņi sql korekti nenostrādā, tad `ps.execute()` izmetīs kļūdu, līdz ar ko, `conn.close()` nemaz neizpildīsies.
Šajā gadījumā, gan īsti nekādas izmaiņās, bet in general, būtu jādara, tā kā rakstīju! :)

Reply:
Šis bija bugs, palicis no veciem laikiem - bija close arī final blokā.